tests/unittests/Types_test.cpp

```diff
@@ -72,6 +72,14 @@ TEST(AVariant, CanBeConstructedFromAComplexValue)
     ASSERT_NO_THROW(sdbus::Variant{value});
 }
 
+TEST(AVariant, CanBeConstructedFromAnStdVariant)
+{
+    using VariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
+    VariantType value{ std::vector<sdbus::Struct<std::string, double>>{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}} };
+
+    ASSERT_NO_THROW(sdbus::Variant{value});
+}
```

Comment:
I would make it stricter and express explicitly that the resulting `sdbus::Variant` does not contain the `value`, but the underlying value of `value`:

```suggestion
TEST(AVariant, CanBeConstructedFromAnStdVariant)
{
    std::vector<sdbus::Struct<std::string, double>> value{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}};
    using StdVariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
    StdVariantType stdVariant{value};

    sdbus::Variant sdbusVariant(value);

    ASSERT_TRUE(variant.containsValueOfType<std::vector<sdbus::Struct<std::string, double>>>());
    ASSERT_THAT(variant.get<std::vector<sdbus::Struct<std::string, double>>>, Eq(value));
}
```

What do you think?

Reply:
I assume here you mean `sdbus::Variant sdbusVariant{stdVariant}`?
Otherwise we aren't testing the new constructor.

Reply:
I think the tests are better this way compared to what I had before.

Reply:
> I assume here you mean `sdbus::Variant sdbusVariant{stdVariant}`? Otherwise we aren't testing the new constructor.

Yes, that's what I meant actually.

> I wonder if we need the extra ASSERT_TRUE.

Not from the verification perspective. I kept it as an explicit statement for future readers (since tests are an up-to-date documentation of behaviors) that the type underneath the variant is vector, not variant. But feel free to remove it if you consider that to be clear from the second assert.

Reply:
No if it's just to be more explicit about the intention I think it's fine to keep it.